src/microbe_stage/systems/SiderophoreSystem.cs

```diff
@@ -0,0 +1,137 @@
+﻿namespace Systems;
+
+using System;
+using Components;
+using DefaultEcs;
+using DefaultEcs.System;
+using DefaultEcs.Threading;
+using World = DefaultEcs.World;
+
+/// <summary>
+///   Handles siderophore projectile collisions
+/// </summary>
+[With(typeof(CollisionManagement))]
+[With(typeof(Physics))]
+[With(typeof(TimedLife))]
+[With(typeof(SiderophoreProjectile))]
+[RunsAfter(typeof(PhysicsCollisionManagementSystem))]
+[RuntimeCost(0.5f, false)]
+public sealed class SiderophoreSystem : AEntitySetSystem<float>
+{
+    /// <summary>
+    ///   Holds a persistent instance of the collision filter callback to not need to create multiple delegates, and
+    ///   to make doubly sure this callback won't be garbage collected while the native side still has a reference to
+    ///   it.
+    /// </summary>
+    private readonly IWorldSimulation worldSimulation;
+
+    public SiderophoreSystem(World world, IParallelRunner runner, IWorldSimulation worldSimulation) :
+        base(world, runner)
+    {
+        this.worldSimulation = worldSimulation;
+    }
+
+    protected override void Update(float delta, in Entity entity)
+    {
+        if (!entity.Has<SiderophoreProjectile>())
+            return;
+
+        ref var projectile = ref entity.Get<SiderophoreProjectile>();
+
+        if (projectile.IsUsed)
+            return;
+
+        ref var collisions = ref entity.Get<CollisionManagement>();
+
+        if (!projectile.ProjectileInitialized)
+        {
+            projectile.ProjectileInitialized = true;
+
+            collisions.StartCollisionRecording(Constants.MAX_SIMULTANEOUS_COLLISIONS_TINY);
+        }
+
+        // Check for active collisions that count as a hit and use up this projectile
+        var count = collisions.GetActiveCollisions(out var activeCollisions);
+        for (int i = 0; i < count; ++i)
+        {
+            ref var collision = ref activeCollisions![i];
+
+            if (!HandleSiderophoreCollision(ref collision, in worldSimulation, entity
+                    .Get<SiderophoreProjectile>().Sender))
+            {
+                continue;
+            }
+
+            // Applied a damaging hit, destroy this toxin
+            // TODO: We should probably get some *POP* effect here.
+
+            // Expire right now
+            ref var timedLife = ref entity.Get<TimedLife>();
+            timedLife.TimeToLiveRemaining = -1;
+
+            ref var physics = ref entity.Get<Physics>();
+
+            // TODO: should this instead of disabling the further collisions be removed from the world immediately
```

Comment:
These TODOs that are duplicated from the toxin system, I think should just be removed. Unless you want to keep the pop effect TODO if you think that the siderophore on collision should have such an effect.

Reply:
This comment hasn't been addressed.